spring-vault-core/src/main/java/org/springframework/vault/core/lease/domain/Lease.java

```diff
@@ -42,14 +44,14 @@ private Lease(String leaseId, long leaseDuration, boolean renewable) {
 	/**
 	 * Create a new {@link Lease}.
 	 *
-	 * @param leaseId must not be empty or {@literal null}.
+	 * @param leaseId must not {@literal null}. Empty string implies a generic secret.
```

Comment:
I would like to keep this method as-is and rather introduce a new factory method just accepting `leaseDuration`. I'm not decided on a name yet, something expressing the source as cache duration hint (`fromTtl`, `fromCacheHint`, …) might be a good fit. WDYT?